image_writers_guide/sti.adoc

```diff
@@ -10,70 +10,52 @@
+- sources
+- STI scripts
+- builder image
 
+During the build process, STI must place sources and scripts inside the builder image. To do so, STI creates a tar file containing the two, then streams that file into the builder image. Before executing the `sti assemble` script, STI untars that file and places its contents into the location specified with the `--location` flag or the `STI_LOCATION` environment variable from the builder image (the default location is [filename]#/tmp#).
 
-== Required image contents
+For this tar process to happen, your image must supply the tar archiving utility (the `tar` command available in [filename]#$PATH#) and the command line interpreter (the `/bin/sh` command); this allows your image to use the fastest possible build path. If the `tar` or `/bin/sh` command is not available, the `sti build` script is forced to automatically perform an additional Docker build to put both the sources and the scripts inside the image, and only then run the usual `sti build` procedure.
 
-The build process consists of three fundamental elements which are combined into
-final docker image, the three are: sources, scripts and builder image. During the
-build process sti must place sources and scripts inside that builder image. To do
-so sti creates a tar file containing the two and then streams that file into the
-builder image. Before executing `assemble` script, sti untars that file and places
-its contents into the location specified with `--location` flag or `STI_LOCATION`
-environment variable from the builder image (default location is `/tmp`). For this
-to happen your image must supply tar archiving utility (command `tar` available in `$PATH`)
-and command line interpreter (command `/bin/sh`). Doing so will allow your image to
-use the fastest possible build path, because in all other cases when either
-`tar` or `/bin/sh` is not available, sti build will be forced to perform an additional
-docker build to put both sources and scripts inside the image and only then run the
-usual sti build procedure (sti will do this automatically if tar and /bin/sh are not found).
-See the following diagram for how build workflow looks like:
+See the following diagram for the basic STI build workflow:
 
-image:sti-flow.png[STI workflow]
+.Build workflow
+image::sti-flow.png[STI workflow]
 
-\* Run build's responsibility is to untar the sources, scripts and artifacts (if such
-exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh`
-error) it's responsible only for invoking `assemble` script, since both scripts and
-sources are already there.
+////
+* Run build's responsibility is to untar the sources, scripts and artifacts (if such exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh` error) it's responsible only for invoking `assemble` script, since both scripts and sources are already there.
+////
 
+== STI Scripts
+STI expects you to supply the following scripts:
```

Comment:
[Suggestion]
You must supply the following scripts to STI:

Even with the suggestion above, does STI really "expect you to supply the following scripts" if some of them are marked "optional".